src/OrbitBase/TemporaryFileTest.cpp

```diff
@@ -46,4 +46,24 @@ TEST(TemporaryFile, Destruction) {
   EXPECT_FALSE(error) << error.message();
 }
 
+TEST(TemporaryFile, CleanupAfterReopen) {
+  std::filesystem::path file_path_copy;
+  {
+    TemporaryFile tmp_file;
+    EXPECT_TRUE(tmp_file.fd().valid());
+    EXPECT_THAT(tmp_file.file_path().string(), HasSubstr("orbit_"));
+    EXPECT_THAT(tmp_file.file_path().string(), Not(HasSubstr("XXXXXX")));
+    file_path_copy = tmp_file.file_path();
+
+    // Check that the file removed even if we call RemoveAndClose and then recreate it.
+    tmp_file.CloseAndRemove();
+    auto fd_or_error = OpenFileForWriting(file_path_copy);
+    ASSERT_TRUE(fd_or_error.has_value()) << fd_or_error.error().message();
```

Comment:
This kind of supports my comment that having both `CloseAndRemove` and the destructor do the same thing in a RAII-like class is confusing. This is a bit weird.

Reply:
The primary goal for this class is to be used in tests. The scope here is very much intentional.